dev/tests/api-functional/testsuite/Magento/GraphQl/ConfigurableProduct/UpdateConfigurableCartItemsTest.php

```diff
@@ -0,0 +1,144 @@
+     */
+    private $quoteFactory;
+
+    /**
+     * @var QuoteResource
+     */
+    private $quoteResource;
+
+    /**
+     * @magentoApiDataFixture Magento/ConfigurableProduct/_files/quote_with_configurable_product.php
+     */
+    public function testUpdateConfigurableCartItemQuantity()
+    {
+        $reservedOrderId = 'test_cart_with_configurable';
+        $maskedQuoteId = $this->getMaskedQuoteId($reservedOrderId);
+        $productSku = 'simple_10';
+        $newQuantity = 123;
+        $quoteItem = $this->getQuoteItemBySku($productSku, $reservedOrderId);
+
+        $query = $this->getQuery($maskedQuoteId, (int)$quoteItem->getId(), $newQuantity);
+        $response = $this->graphQlMutation($query);
+
+        self::assertArrayHasKey('updateCartItems', $response);
+        self::assertArrayHasKey('quantity', $response['updateCartItems']['cart']['items']['0']);
+        self::assertEquals($newQuantity, $response['updateCartItems']['cart']['items']['0']['quantity']);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->getMaskedQuoteIdByReservedOrderId = $objectManager->get(GetMaskedQuoteIdByReservedOrderId::class);
+        $this->quoteFactory = $objectManager->get(QuoteFactory::class);
+        $this->quoteResource = $objectManager->get(QuoteResource::class);
+        $this->quoteIdMaskFactory = Bootstrap::getObjectManager()->get(QuoteIdMaskFactory::class);
+    }
+
+    /**
+     * @param string $maskedQuoteId
+     * @param int $quoteItemId
+     * @param int $newQuantity
+     * @return string
+     */
+    private function getQuery(string $maskedQuoteId, int $quoteItemId, int $newQuantity): string
+    {
+        return <<<QUERY
+mutation {
+  updateCartItems(input: {
+    cart_id:"$maskedQuoteId"
+    cart_items: [
+      {
+        cart_item_id: $quoteItemId
+        quantity: $newQuantity
+      }
+    ]
+  }) {
+    cart {
+      items {
+        quantity
+      }
+    }
+  }
+}
+QUERY;
+    }
+
+    /**
+     * Returns quote item by product SKU
+     *
+     * @param string $sku
+     * @return Item|bool
+     * @throws NoSuchEntityException
+     */
+    private function getQuoteItemBySku(string $sku, string $reservedOrderId)
+    {
+        $quote = $this->quoteFactory->create();
+        $this->quoteResource->load($quote, $reservedOrderId, 'reserved_order_id');
+        $item = false;
+        foreach ($quote->getAllItems() as $quoteItem) {
+            if ($quoteItem->getSku() == $sku && $quoteItem->getProductType() == Configurable::TYPE_CODE &&
+                !$quoteItem->getParentItemId()) {
+                $item = $quoteItem;
+                break;
+            }
+        }
+
+        return $item;
+    }
+
+    /**
+     * @param $reservedOrderId
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    private function getMaskedQuoteId(string $reservedOrderId): string
```

Comment:
Hi @vovsky. Feel free to use the built-in service for this purpose. Some of the reusable logic has been decoupled to separate services.

Reply:
Unfortunately I can not use that service because fixture that I use (Magento/ConfigurableProduct/_files/quote_with_configurable_product.php) doesn't create quote mask (record in quote_id_mask table)

Reply:
Consider adding masked quote id to this fixture.

Reply:
@lenaorobei quote mask generation has been added to fixture, please review. Thank you